### PR TITLE
RDKTV-20401: TV did not enter to Deepsleep , and stayed ~14 hrs in th…

### PR DESCRIPTION
**Pull request by PaviP98**
…e standby state

Maintenance window is open for long time and it was not completed successfully and hence TV didn't went to deepsleep.

Whenever stopMaintenance Api executed, it will terminate the task which is currently running and will send that event status to MM, also it will update the status of remaining tasks as well. Since MM didn't trigger those scripts it will simply ignore those events and so maintenance was not completed. Hence added changes to directly send the MAINTENANCE_ERROR event from stopMaintenance()

### CODE REVIEW

File: MaintenanceManager/MaintenanceManager.cpp [hunk 1]
```diff
@@ -1287,7 +1286,7 @@ namespace WPEFramework {
                                 if (k_ret == 0){
                                     LOGINFO(" %s Termimated\n",script_names[i].c_str());
                                     /*this means we killed the task currently running */
-                                    task_incomplete = true;
+                                    m_task_map[task_names_foreground[i].c_str()]=false;
                                 }
```
Comment (mvalas827) [inline MaintenanceManager/MaintenanceManager.cpp:1290]: Shouldn't we break the loop here as we only run one task at a time and abort flag update should stop running any following scripts in line.

File: MaintenanceManager/MaintenanceManager.cpp [hunk 2]
```diff
@@ -1287,7 +1286,7 @@ namespace WPEFramework {
                                 if (k_ret == 0){
                                     LOGINFO(" %s Termimated\n",script_names[i].c_str());
                                     /*this means we killed the task currently running */
-                                    task_incomplete = true;
+                                    m_task_map[task_names_foreground[i].c_str()]=false;
```
Comment (mvalas827) [inline MaintenanceManager/MaintenanceManager.cpp:1289]: m_task_map is being updated from two threads.. plz check on possible race conditions.

**Review by mvalas827 — COMMENTED** [review]
Test the change thoroughly by aborting at various intervals in different scenarios.

I see notify_status and m_notify_status not in sync. Try to include it as part of cleanup if you see any issues with it.

File: MaintenanceManager/MaintenanceManager.cpp (hunk 2 above)
Comment (PaviP98) [reply MaintenanceManager/MaintenanceManager.cpp:1289]: m_task_map is updated from stopMaintenance only. Test results are attached in the ticket https://ccp.sys.comcast.net/browse/RDKTV-20401?focusedCommentId=15064916&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-15064916. Verified both unsolicited and solicited maintenance by invoking stopMaintenance api. No crashes or regressions observed with Maintenance Manager

File: MaintenanceManager/MaintenanceManager.cpp [hunk 3]
```diff
@@ -862,6 +872,8 @@ namespace WPEFramework {
             else {
                 LOGWARN("Ignoring unexpected event - owner: %s, eventId: %d!!", owner, eventId);
             }
+          }
```
Comment (tdeva14) [inline MaintenanceManager/MaintenanceManager.cpp:875]: Can we have a log statement here to indicate that abort flag has been set and hence ignoring the event ?

Comment (PaviP98) [reply MaintenanceManager/MaintenanceManager.cpp:875]: Done

**Review by tdeva14 — COMMENTED** [review]
Changes look good

**Review by mvalas827 — APPROVED** [review]
Looks good. Thanks

**Review by anand-ky — APPROVED** [review]

**Pull request merged**